gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinApplicationMaster.java

```diff
@@ -229,6 +227,8 @@ public static void main(String[] args) throws Exception {
         printUsage(options);
         System.exit(1);
       }
+      Config config = ConfigFactory.load();
```

Comment:
Also it would be good to add a comment explaining why the token update is done in the main() i.e "Because AM is restarted with the original AppSubmissionContext, it may have outdated delegation tokens. So the refreshed tokens should be added into the container's UGI before any HDFS/Hive/RM access is performed."